nginx/config.go

```diff
@@ -73,8 +73,9 @@ http {
 	# treated as if it were HTTPs.
 	map $http_x_forwarded_proto $tmp_access_scheme {
 		default $scheme;               # if X-Forwarded-Proto header is empty, $tmp_access_scheme will be the actual protocol used
-		"~^(.*, ?)?http$" "http";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value
-		"~^(.*, ?)?https$" "https";    # account for the possibility of a comma-delimited X-Forwarded-Proto header value
+		"~^(.*, ?)?https(,.* ?)?$" "https";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value to see if https is present
+		"~^(.*, ?)?http(,.* ?)?$" "http";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value to see if http is present
+		"~^[A-Za-z][A-Za-z0-9+.-]*" $http_x_forwarded_proto; # Use the value passed in the header if its a valid URI scheme
```

Comment:
the regex is based on the rfc for a valid scheme name https://www.ietf.org/rfc/rfc2396.txt(section 3.1) and yes it accepts all the scheme names both which are accepted by IANA http://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml and the private schemes.
so here my question is do we want to be loose on what needs to be accepted or more strict and add/handle them upon user requests(currently the request is only for web sockets)?
i had actually forgot about the EnforceSSL part, websockets have two schemes ws:// and wss:// similar to http and if enforcessl is present we need to change the scheme from ws to wss.